src/bridge/cockpitdbuscache.c

```diff
@@ -529,11 +529,18 @@ introspect_queue (CockpitDBusCache *self,
   id->callback = callback;
   id->user_data = user_data;
 
-  g_debug ("%s: queueing introspect %s %s%s", self->logname, path,
-           interface ? "for " : "", interface ? interface : "");
-  g_queue_push_tail (self->introspects, id);
-
-  introspect_next (self);
+  if (!g_cancellable_is_cancelled (self->cancellable))
+    {
+      g_debug ("%s: queueing introspect %s %s%s", self->logname, path,
+               interface ? "for " : "", interface ? interface : "");
+      g_queue_push_tail (self->introspects, id);
+      introspect_next (self);
+    }
+  else
+    {
+      g_debug ("%s: cancelling queueing introspect %s %s%s", self->logname, path,
```

Comment:
Hmmm, it seems like id leaks as does user_data. Should we instead short the actual Introspect call during dispose, rather than skipping at this point?